token_update_nfts_transaction_e2e_test.go

```diff
@@ -0,0 +1,317 @@
+	_, err = updateNftMetadataWithWrongKey(t, &env, tokenID, nftSerials, updatedMetadata, env.OperatorKey)
+	require.Error(t, err)
+}
+
+func TestCannotUpdateNFTMetadataWhenMetadataKeyWasRemoved(t *testing.T) {
+	env := NewIntegrationTestEnv(t)
+
+	// Generate metadata key
+	metadataKey, err := PrivateKeyGenerateEd25519()
+	require.NoError(t, err)
+
+	nftCount := 4
+	initialMetadataList := generateNftMetadata([]byte{4, 2, 0}, nftCount)
+	updatedMetadata := []byte{6, 9}
+
+	// Create a token with a metadata key
+	tokenID := createTokenWithMetadataKey(t, &env, &metadataKey)
+
+	// Check metadata key before removal
+	tokenInfoBeforeUpdate := getTokenInfo(t, &env, tokenID)
+	assert.Equal(t, metadataKey.PublicKey().String(), tokenInfoBeforeUpdate._ToProtobuf().MetadataKey.String(), "metadata key should match before removal")
+
+	// Remove metadata key
+	updateTokenMetadataKey(t, tokenID, nil, &env)
+
+	// Check metadata key after removal
+	tokenInfoAfterUpdate := getTokenInfo(t, &env, tokenID)
+	assert.Nil(t, tokenInfoAfterUpdate._ToProtobuf().MetadataKey, "metadata key should be nil after removal")
+
+	// Mint tokens
+	tokenMintTransactionReceipts := mintTokens(t, &env, tokenID, initialMetadataList)
+	nftSerials := make([]int64, 0)
+	for _, receipt := range tokenMintTransactionReceipts {
+		nftSerials = append(nftSerials, receipt.SerialNumbers[0])
+	}
+
+	// Check NFTs' metadata can't be updated when a metadata key is not set
+	_, err = updateNftMetadataWithNoMetadataKey(t, &env, tokenID, nftSerials, updatedMetadata, metadataKey)
+	require.Error(t, err)
+}
+
+// Utility functions
+
+func createTokenWithMetadataKey(t *testing.T, env *IntegrationTestEnv, metadataKey *PrivateKey) TokenID {
+	tokenCreateTx := NewTokenCreateTransaction().
+		SetNodeAccountIDs(env.NodeAccountIDs).
+		SetTokenName("ffff").
+		SetTokenSymbol("F").
+		SetTokenType(TokenTypeNonFungibleUnique).
+		SetTreasuryAccountID(env.Client.GetOperatorAccountID()).
+		SetAdminKey(env.Client.GetOperatorPublicKey()).
+		SetSupplyKey(env.Client.GetOperatorPublicKey()).
+		SetMetadataKey(metadataKey)
+
+	tx, err := tokenCreateTx.Execute(env.Client)
+	require.NoError(t, err)
+
+	receipt, err := tx.GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	return *receipt.TokenID
+}
+
+func createTokenWithoutMetadataKey(t *testing.T, env *IntegrationTestEnv) TokenID {
+	tokenCreateTx := NewTokenCreateTransaction().
+		SetNodeAccountIDs(env.NodeAccountIDs).
+		SetTokenName("ffff").
+		SetTokenSymbol("F").
+		SetTokenType(TokenTypeNonFungibleUnique).
+		SetTreasuryAccountID(env.Client.GetOperatorAccountID()).
+		SetAdminKey(env.Client.GetOperatorPublicKey()).
+		SetSupplyKey(env.Client.GetOperatorPublicKey())
+
+	tx, err := tokenCreateTx.Execute(env.Client)
+	require.NoError(t, err)
+
+	receipt, err := tx.GetReceipt(env.Client)
+	require.NoError(t, err)
+
+	return *receipt.TokenID
+}
+
+func updateTokenMetadataKey(t *testing.T, tokenID TokenID, metadataKey *PrivateKey, env *IntegrationTestEnv) *TokenID {
```

Comment:
return also the error and assert it in the specific test case, remove the other variants of this method